### PR TITLE
Overcommit with rubocop setup

### PR DESCRIPTION
**Pull request by DevKaranPradhan**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+Layout/LineLength:
+ Max: 250
+
```
Comment (tanya-saroha) [inline .rubocop.yml]: @DevKaranPradhan  Why 250?

File: Gemfile [hunk 2]
```diff
@@ -19,29 +21,30 @@ gem 'puma', '~> 3.11'
...
+gem "bootsnap", ">= 1.1.0", require: false
 
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
 # gem 'rack-cors'
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem "byebug", platforms: %i[mri mingw x64_mingw]
+  gem "overcommit", "~> 0.52.1"
```
Comment (tanya-saroha) [inline Gemfile]: @DevKaranPradhan  Whenever you add new gems, it is a good practice to add a short comment stating the reason of adding that gem.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+Metrics/MethodLength:
+ Max: 20
```
Comment (tanya-saroha) [inline .rubocop.yml]: Why 20?

File: .rubocop.yml (hunk 1 above)
Comment (DevKaranPradhan) [reply .rubocop.yml]: mam, there was a line in schema.rb which was over 70. what should I set it to?

**Comment by DevKaranPradhan** [comment]
[Integrate overcommit with rubocop](https://trello.com/c/X1AqpiHc/29-integrate-overcommit-with-rubocop)

File: .rubocop.yml (hunk 1 above)
Comment (tanya-saroha) [reply .rubocop.yml]: Please dont use mam

**Pull request merged**